fix: add TypeScript type definition for request.earlyResponse

iiroj opened the pull request:
This PR adds TypeScript type definitions for the `Request.earlyResponse` field. Fixes issue https://github.com/middyjs/middy/issues/1275

willfarrell commented:
Thanks for the PR, I think it needs a small update

```
packages/util/index.test-d.ts:22:6
  ✖  22:6  Property earlyResponse is missing in type { event: { body: string; headers: {}; multiValueHeaders: {}; httpMethod: string; isBase64Encoded: false; path: string; pathParameters: {}; queryStringParameters: {}; multiValueQueryStringParameters: {}; stageVariables: {}; requestContext: { ...; }; resource: string; }; context: { ...; }; response: null; error: null;... but required in type Request<APIGatewayProxyEvent, APIGatewayProxyResult, Error, Context, TInternal>.  
  ```
 
  ```
  /home/runner/work/middy/middy/packages/core/index.test-d.ts:429:16: Unnecessary parentheses around expression. (@typescript-eslint/no-extra-parens)
```

iiroj commented:
I'll fix the PR tomorrow now that the release issue is sorted out!

iiroj commented:
@willfarrell ok the PR should now be now good, although I got into a bit of a traffic hole with the development setup:

1. After running `npm install` I got this diff in `package-lock.json`: https://github.com/middyjs/middy/pull/1277/commits/c0e6d72e4a30aec69838af594b5af78bc39a08e8
2. I noticed Husky wasn't set to be automatically setup, so I added the `prepare` script: https://github.com/middyjs/middy/pull/1277/commits/3f71cef4977283e1178674b63657f3e1b53c7445
3. The TypeScript files weren't being linted during commit, so I added them: https://github.com/middyjs/middy/pull/1277/commits/d6e0dbaeea8b136aa5b3828a74536e3a85ac24e8
4. This revealed some formatting changes to the existing test file: https://github.com/middyjs/middy/pull/1277/commits/da16876fda197fab75cffd0efbd73c79748bbf9f
5. Finally, the fix itself: https://github.com/middyjs/middy/pull/1277/commits/f0dabe5c213194effc3709339ba1c071b847e81e

iiroj commented:
Btw, I think the current `prettier` and `ts-standard` formatters are conflicting with each other a bit.

willfarrell commented:
Amazing stuff. Thanks!

> I think the current prettier and ts-standard formatters are conflicting with each other a bit.

I'm aware, kind of a legacy thing. I've added it to the v7 release to be updated.
